src/main/scala/day06.scala

```diff
@@ -0,0 +1,19 @@
+// Option is a sum type. Create you own Option type, call it MyOption. You should have a case for Some and a case for None
+sealed trait MyOption[+A]
+final case class MySome[+A](a: A) extends MyOption[A]
+case object MyNone extends MyOption[Nothing]
+
+// Create a function that takes MyOption as an argument. If it contains a value, it should print it, otherwise it should print "None"
+val print: (MyOption[String]) => Unit = {
```

Comment:
This doesn't work because a `val` definition needs to be scoped inside either a class or an object definition.

```suggestion
object day06 {
  // Create a function that takes MyOption as an argument. If it contains a value, it should print it, otherwise it should print "None"
  val print: MyOption[String] => Unit = {
    case MyNone => println("None")
    case MySome(t) => println(t)
  }

  def main(args: Array[String]): Unit = {
    val myOption = MySome("myoption")
    println(myOption)
  }
}
```